### PR TITLE
Spring SecurityFilterChain related recipes remove antmatchers rules

### PR DESCRIPTION
**Pull request by Laurens-W**
## What's changed?
Added additional test cases to show an issue exists

## What's your motivation?
- Fixes https://github.com/openrewrite/rewrite-spring/issues/545
- Fixes https://github.com/openrewrite/rewrite-spring/issues/529
- Fixes https://github.com/openrewrite/rewrite-spring/issues/463

## Any additional context
- https://github.com/openrewrite/rewrite-spring/pull/419

### CODE REVIEW

File: src/testWithSpringSecurity_5_7/java/org/openrewrite/spring/security5/UpgradeSpringSecurity61Test.java [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+import java.util.Collection;
+import java.util.List;
+
+import static org.openrewrite.java.Assertions.java;
+
+class UpgradeSpringSecurity61Test implements RewriteTest {
+
+    @Override
+    public void defaults(RecipeSpec spec) {
+        spec.recipeFromResources("org.openrewrite.java.spring.security6.UpgradeSpringSecurity_6_1")
```
Comment (timtebeek) [inline src/testWithSpringSecurity_5_7/java/org/openrewrite/spring/security5/UpgradeSpringSecurity61Test.java:36]: Can we narrow this down to one particular recipe that's making the incorrect change? Or a combination of two that does not run through the whole migration? Ideally if we find just one we move the tests to the respective test class.

File: src/testWithSpringSecurity_5_7/java/org/openrewrite/spring/security5/UpgradeSpringSecurity61Test.java [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+              @EnableWebSecurity
+              class SecurityConfig extends WebSecurityConfigurerAdapter {
+                  @Override
+                  public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
+                      http
+                          .authorizeRequests()
+                          .antMatchers(HttpMethod.OPTIONS, "/rest/**").permitAll()
+                          .anyRequest().permitAll()
+                          .and().csrf().disable();
+                      return http.build();
```
Comment (timtebeek) [inline src/testWithSpringSecurity_5_7/java/org/openrewrite/spring/security5/UpgradeSpringSecurity61Test.java:65]: Is there anything we can strip from this block while still replicating the issue? Perhaps the csrf is optional? Or is it not?

File: src/testWithSpringSecurity_5_7/java/org/openrewrite/spring/security5/UpgradeSpringSecurity61Test.java [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+                      http
+                          .authorizeRequests()
+                          .antMatchers(HttpMethod.OPTIONS, "/rest/**").permitAll()
+                          .antMatchers("/openapi").permitAll()
+                          .antMatchers("/openapi.yaml").permitAll()
+                          .antMatchers("/openapi/**").permitAll()
+                          .antMatchers("/rest/hello").permitAll()
+                          .antMatchers("/actuator/**").permitAll()
+                          .anyRequest().authenticated()
+                          .and().oauth2ResourceServer().jwt();
```
Comment (timtebeek) [inline src/testWithSpringSecurity_5_7/java/org/openrewrite/spring/security5/UpgradeSpringSecurity61Test.java:129]: Similar to above; can we strip out anything not necessary to replicate the issue? Eyeing the `oauth2ResourceServer` here, possibly others.

**Comment by Laurens-W** [comment]
I think it's safe to say the combination of the order of recipes and the type information being broken by the `UseNewRequestMatchers` causing repeated cycles caused the side effects described in the 3 issues linked

**Review by timtebeek — APPROVED** [review]
Great to see this resolved finally! Thanks for diving in and uncovering that missing method type change.

**Pull request merged**